task/dump-agent/src/main.rs

```diff
@@ -125,8 +125,28 @@ impl idl::InOrderDumpAgentImpl for ServerImpl {
             &area.address.to_le_bytes(),
             &mut buf,
         ) {
-            Err(_) => Err(DumpAgentError::DumpFailed.into()),
-            Ok(_) => Ok(()),
+            Err(_) => Err(DumpAgentError::DumpMessageFailed.into()),
+            Ok(result) => {
```

Comment:
What would you feel about better comments here?  To me (and despite having a bias towards early returns!), I like the nesting because it tells the reader that every one of these paths is a return path -- but I think some comments would make it clearer.

Reply:
I mean, I think the fact that this is the end of the method makes it clear that they're all return paths. :) The thing that threw me the most was the `if val != 0` bit; if just that were flattened I'd be pretty happy:

```rust
        if val == 0 {
            Ok(())
        } else if let Some(err) = dumper_api::DumperError::from_u32(val) {
            Err(DumpAgentError::from(err).into())
        } else {
            Err(DumpAgentError::DumpFailedUnknownError.into())
        }
```

Reply:
Agreed -- that's definitely an improvement! Adding a comment as well.